sdk/core/Azure.Core/tests/TestFramework/SimulateRequestRetryAttribute.cs

```diff
@@ -0,0 +1,34 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Threading;
+using NUnit.Framework;
+using NUnit.Framework.Interfaces;
+using NUnit.Framework.Internal;
+
+namespace Azure.Core.Testing
+{
+    [AttributeUsage(AttributeTargets.Method | AttributeTargets.Class | AttributeTargets.Assembly, Inherited = false)]
+    public class SimulateRequestRetryAttribute : NUnitAttribute, ITestAction
```

Comment:
I think an attribute for this task is overkill. Can we have this feature turned on by property on RecordedTestBase?

Reply:
Or parameter on `InstrumentOptions`?

Reply:
@tg-msft , @JoshLove-msft Storage has its own `ResponseClassifier`. Do you need to be able to customize retry simulation for it?

Reply:
We use a [pipeline policy](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/storage/Azure.Storage.Common/tests/Shared/TestExceptionPolicy.cs) for simulating failures. But we only add this for tests where we are explicitly testing retries. Is the idea here that we can just add an attribute to any existing tests and get retry coverage for free? If so, I think we would want to use this in the future. What would need to be updated to hook in our StorageResponseClassifier?

Reply:
Would be nice to unify implementations as part of this PR.

Reply:
Then I should probably expand it to a more general `FailureSimulationTransport` (as @pakrym suggested).